### ISSUE TITLE
[no-duplicate-imports]: false positive when import and import type

### ISSUE THREAD

**Issue by ghaiklor-wix** [eslint/eslint#13669]
**Tell us about your environment**

Node version: v12.18.3
npm version: v6.14.6
Local ESLint version: v7.8.0

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
  env: {
    commonjs: true,
    es6: true,
    jest: true,
    node: true,
  },
  extends: [
    'eslint:all',
    'plugin:@typescript-eslint/all',
    'plugin:import/errors',
    'plugin:import/warnings',
    'plugin:import/typescript',
    'plugin:node/recommended',
    'plugin:promise/recommended',
    'standard-with-typescript',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.eslint.json',
    tsconfigRootDir: __dirname,
  },
  plugins: [
    '@typescript-eslint',
    'import',
    'node',
    'promise',
    'standard',
  ],
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import type { TestkitServer } from 'testkit';
import { initTestkit } from 'testkit';
```

```bash
eslint --fix .
```

**What did you expect to happen?**

There is another rule that auto-fixes imports to have both `import type` and `import`. TypeScript can use `import type` to somehow optimize its inner "things".

I expect to not have a duplicate report when `import type` and `import` is used.

**What actually happened? Please include the actual, raw output from ESLint.**

```
'testkit' import is duplicated
```

**Are you willing to submit a pull request to fix this bug?**

Possibly

**Comment by btmills** [comment]
@ghaiklor-wix can you update the issue with more details on ESLint's output? I can't currently tell what rule is reporting the duplicate import and whether it's an ESLint core rule or a plugin. If I had to guess, I'd say this is most likely a configuration issue. ESLint's core rules don't know about TypeScript's syntax extensions, so the TS plugin provides custom implementations for those rules. The core rules should be disabled where there are type-aware replacements available.

**Comment by ghaiklor-wix** [comment]
@btmills you are right, so here is why it is happening...

typescript-eslint plugin has a rule [consistent-type-imports](https://github.com/typescript-eslint/typescript-eslint/blob/v4.0.1/packages/eslint-plugin/docs/rules/consistent-type-imports.md).

> TypeScript 3.8 added support for type-only imports. Type-only imports allow you to specify that an import can only be used in a type location, allowing certain optimizations within compilers.

This rule has an auto-fix that splits imports into two separate categories `import type` and `import`. Which is conflicts with this rule.

For now, I've disabled this one rule, but maybe there is a way to configure it with something like `allowImportType`?

**Comment by btmills** [comment]
[`import/no-duplicates`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-duplicates.md) is already aware of `type` imports. Looks like you might be using that already? If you disable the core rule like you've already done and use `import/no-duplicates` instead, you should be good to go.

**Comment by mdjermanovic** [comment]
There is the same issue in the typescript-eslint repo: https://github.com/typescript-eslint/typescript-eslint/issues/2315

Looks like it still isn't clear whether a new rule will be added to `@typescript-eslint/eslint-plugin`, or the existing `import/no-duplicates` rule already does the work.

Either way, the core `no-duplicate-imports` doesn't support this typescript syntax, so a replacement rule should be used instead. It could be useful to follow https://github.com/typescript-eslint/typescript-eslint/issues/2315.

**Comment by ghaiklor-wix** [comment]
So the solution here is basically to disable the rule and delegate the work to `import/no-duplicates`.

**Comment by ghaiklor-wix** [comment]
Yeah, I'm extending as many as I can. Kind of people who enable everything and just configure it in its own way as I like it. So, that's pretty sums everything up and we can close the issue.

Instead of using core rule, better to use `import/no-duplicates`.

**Comment by TokugawaTakesi** [comment]
Please tell me why you closing the issues without at least announcing of fix realize.

**Comment by mdjermanovic** [comment]
Hi @TokugawaTakesi!

ESLint core (built-in) rules are made to fully support ECMAScript syntax. Most of the core rules work perfectly well with TypeScript as well, but some don't.

Per our [policy](https://github.com/eslint/eslint#what-about-experimental-features), we may make changes to core rules to better work with TypeScript on a case-by-case basis. We're usually doing that if it is a small change that doesn't require the rule to be aware of TypeScript syntax extensions. 

The core `no-duplicate-imports` rule can't support `import type` syntax. In such cases, we recommend using a rule from a plugin instead (usually an [extension rule](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules) from [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin)).

Please follow typescript-eslint/typescript-eslint#2315. I believe the conclusion is to use `import/no-duplicates` rule, but the issue isn't closed yet.

**Comment by ghaiklor-wix** [comment]
@TokugawaTakesi if you inspect to our discussion here, you find that the fix is already present here.

> import/no-duplicates is already aware of type imports. Looks like you might be using that already? If you disable the core rule like you've already done and use import/no-duplicates instead, you should be good to go.

> Either way, the core no-duplicate-imports doesn't support this typescript syntax, so a replacement rule should be used instead.

> So the solution here is basically to disable the rule and delegate the work to import/no-duplicates.

**Comment by TokugawaTakesi** [comment]
Dear all, thank you for the respond.

> In such cases, we recommend using a rule from a plugin instead
I am sorry. Looks like I confused when came here by link from @typescript/eslint conversation.
I even did not notice when moved from `@typescript/eslint` repository to `eslint` repository.

> Please follow typescript-eslint/typescript-eslint#2315.
Thank you! I subscribed on this issue.

Sorry for disturbing.
